homeassistant/components/mqtt/lawn_mower.py

```diff
@@ -0,0 +1,233 @@
+"""Support for MQTT lawn mowers."""
+from __future__ import annotations
+
+from collections.abc import Callable
+import contextlib
+import functools
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import lawn_mower
+from homeassistant.components.lawn_mower import (
+    LawnMowerActivity,
+    LawnMowerEntity,
+    LawnMowerEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_OPTIMISTIC, CONF_VALUE_TEMPLATE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import subscription
+from .config import MQTT_BASE_SCHEMA
+from .const import (
+    CONF_COMMAND_TEMPLATE,
+    CONF_COMMAND_TOPIC,
+    CONF_ENCODING,
+    CONF_QOS,
+    CONF_RETAIN,
+    CONF_STATE_TOPIC,
+    CONF_SUPPORTED_FEATURES,
+    DEFAULT_OPTIMISTIC,
+    DEFAULT_RETAIN,
+)
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, MqttEntity, async_setup_entry_helper
+from .models import (
+    MqttCommandTemplate,
+    MqttValueTemplate,
+    PublishPayloadType,
+    ReceiveMessage,
+    ReceivePayloadType,
+)
+from .util import get_mqtt_data, valid_publish_topic, valid_subscribe_topic
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_OPTIONS = "options"
+
+DEFAULT_NAME = "MQTT Lawn Mower"
+ENTITY_ID_FORMAT = lawn_mower.DOMAIN + ".{}"
+
+MQTT_LAWN_MOWER_ATTRIBUTES_BLOCKED: frozenset[str] = frozenset()
+
+_SUPPORTED_FEATURES = {
+    "dock": LawnMowerEntityFeature.DOCK,
+    "pause": LawnMowerEntityFeature.PAUSE,
+    "start_mowing": LawnMowerEntityFeature.START_MOWING,
+}
+
+PLATFORM_SCHEMA_MODERN = MQTT_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_COMMAND_TOPIC): valid_publish_topic,
+        vol.Optional(CONF_COMMAND_TEMPLATE): cv.template,
+        vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
+        vol.Optional(CONF_RETAIN, default=DEFAULT_RETAIN): cv.boolean,
+        vol.Optional(CONF_STATE_TOPIC): valid_subscribe_topic,
```

Comment:
This is called `activity` in the `lawn_mower` entity model, I think it's clearer if we call it that here too.